wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -391,14 +393,23 @@ class KotlinGenerator private constructor(
             if (emitAndroid) {
               newName("CREATOR", "CREATOR")
             }
-            message.fieldsAndOneOfFields.forEach { field ->
+            message.fieldsAndFlatOneOfFields().forEach { field ->
               if (field.name == field.type!!.simpleName ||
-                  schema.getType(field.qualifiedName) != null) {
+                schema.getType(field.qualifiedName) != null
+              ) {
                 newName(field.qualifiedName, field)
               } else {
                 newName(field.name, field)
               }
             }
+            message.boxOneOfs().forEach { oneOf ->
+              newName(oneOf.name, oneOf)
+              newName(oneOf.name + "Keys",oneOf.name + "Keys")
+              newName(oneOf.name.capitalize(), oneOf.name.capitalize())
+               oneOf.fields.forEach { field ->
+                 newName(oneOf.name + field.name.capitalize(), oneOf.name + field.name.capitalize())
+               }
```

Comment:
I think we want a little indirection model here, so we can preserve declaration order.